warehouse/packaging/models.py

```diff
@@ -162,6 +162,11 @@ def __contains__(self, project):
             return True
 
 
+class LifecycleStatus(enum.StrEnum):
+    QuarantineEnter = "quarantine-enter"
+    QuarantineExit = "quarantine-exit"
```

Comment:
Should this include a FSM like we use for `EmailStatus`? https://github.com/pypi/warehouse/blob/15de75987757412a0cb38249ea60697938406c50/warehouse/email/ses/models.py#L40

It might not seem necessary now but I could see it becoming useful if/when we start adding additional statuses.

Reply:
That's an interesting idea - thanks for suggesting! I'm less familiar with automat or the specifics of our implementation.

I'll defer this for now, considering the unknown effort to implement this, and make note for further lifecycle work that we might entertain.
cc @woodruffw as an interested party

Reply:
(Following up from https://github.com/pypi/warehouse/pull/16179#discussion_r1668897815)

I agree with @di about there being value in having this be modeled as a FSM (although it doesn't necessarily have to use `automat`, since it can be an anydirectional machine as long as there's a single "nominal" state. But deferring on that also seems fine, given your point about data migration + unknown effort; just something to consider in the longer run.

(As part of that though, maybe it makes sense to omit `QuarantineExit` and use only `None` as the terminal state? That would make future changes easier I think. But maybe that makes it harder to track the semantics of "leaving quarantine"...)